core/src/main/scala/org/apache/spark/internal/config/UI.scala

```diff
@@ -191,4 +191,14 @@ private[spark] object UI {
     .version("3.0.0")
     .stringConf
     .createOptional
+
+  val MASTER_UI_DECOMMISSION_ALLOW_MODE = ConfigBuilder("spark.master.ui.decommission.allow.mode")
+    .doc("Specifies the behavior of the Master Web UI's /workers/kill endpoint. Possible choices" +
+      " are: `local` means allow this endpoint from IP's that are local to the machine running" +
+      " the Master, `deny` means to completely disable this endpoint, `allow` means to allow" +
+      " calling this endpoint from any IP.")
+    .internal()
+    .version("3.1.0")
+    .stringConf
```

Comment:
it's common to always upper case the config value, as it should be case insensitive. e.g.
```
...
  .stringConf
  .transform(_.toUpperCase(Locale.ROOT))
```